### ISSUE TITLE
[Messenger] Publishing message should not create and bind a queue

### ISSUE THREAD

**Issue by davidbarratt** [symfony/symfony#26937]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 4.1

If you are publishing messages only (and another application is receiving them), then a queue should not be declared/bound.

The queue declaration and binding found here:
https://github.com/symfony/symfony/blob/ccb0e8479901358e7a7cb0c98d3b58990c619386/src/Symfony/Component/Messenger/Adapter/AmqpExt/Connection.php#L156-L157

Should not run within `publish()`
https://github.com/symfony/symfony/blob/ccb0e8479901358e7a7cb0c98d3b58990c619386/src/Symfony/Component/Messenger/Adapter/AmqpExt/Connection.php#L95-L102

The work around to this issue, is to ensure that the application listening for the messages uses the same queue name.

**Comment by sroze** [comment]
In your scenario (i.e. publishing "outside" of the application boundary) you are right, it shouldn't declare anything (even if right now it will only do this in debug mode). The reason we do this is obvious, it's to have a smooth developer experience.

Would a parameter meet your need? Something like this? (or via the `options`)
```yaml
adapters:
    yourname: amqp://localhost/%2f/messages?auto-configuration=false
```

**Comment by davidbarratt** [comment]
yeah if it's configurable in some way, that is fine.

or maybe `declare-queue-on-publish` ?

**Comment by sroze** [comment]
I guess `declare-queue` is a good name, it would apply to both receive and publish.

**Comment by javiereguiluz** [comment]
`declare` is not often used in Symfony and to me the intention is not clear. I don't understand the purpose of this, but please consider using more common names, such as : `create-queue`, `bind-queue`, `unbound-queue`, etc.

**Comment by sroze** [comment]
What about the existing wording in the code ("setup"):
```
amqp://localhost/%2f/messages?auto-setup=false
```

**Comment by sroze** [comment]
You can have a look to #26956.

**Comment by sroze** [comment]
Fixed by #26956.
